autoprecompiles/src/optimizer.rs

```diff
@@ -33,58 +44,109 @@ pub fn optimize<A: Adapter>(
         stats_logger.log("exec bus optimization", &machine);
     }
 
-    let mut constraint_system = symbolic_machine_to_constraint_system(machine);
+    let constraint_system = symbolic_machine_to_constraint_system(machine);
+    let constraint_system = introduce_bus_interaction_variables(constraint_system);
+
+    // Run the optimizer while avoiding inlining bus interaction field variables
+    let constraint_system =
+        run_optimization_loop_until_no_change::<_, _, _, A::MemoryBusInteraction<_>>(
+            constraint_system,
+            bus_interaction_handler.clone(),
+            |var, expr, _system| {
```

Comment:
This still seems pretty ad-hoc to me. Is there any advantage of inlining at all? What happens if you just return `false`?

If it needs to be to be this complicated, I'd vote for extracting a function, similar to `inline_everything_below_degree_bound`.

Reply:
Ah sorry, it stopped the sha256 test after 10 minutes. Usually it takes 2-3